tls/s2n_handshake.h

```diff
@@ -73,6 +74,9 @@ typedef enum {
     END_OF_EARLY_DATA,
 
     APPLICATION_DATA,
+
+    /* Not a standardized message. Defined: https://datatracker.ietf.org/doc/html/draft-agl-tls-nextprotoneg-04 */
+    TLS12_ENCRYPTED_EXTENSIONS,
```

Comment:
I think we've generally been trying to keep APPLICATION_DATA as the last message. I don't remember why tho.